fpga_interchange/static_timing_analysis.py

```diff
@@ -0,0 +1,639 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2021 The SymbiFlow Authors.
+#
+# Use this source code is governed by a ISC-style
+# license that can be found in LICENSE file or at
+# https://opensource.org/licenses/ISC
+#
+# SPDX-License-Identifier: ISC
+"""
+This file defines TimingAnalyzer class.
+
+The TimingAnalyzer is static timing analyzer based on Elmore delay model.
+It takes as an input device, physical netlist and schema path.
+Before running analysis it is required to run all create functions as netlist patcher and
+delay calculator uses maps generated by these functions.
+
+Patching is performed in steps:
+    1. For all trees in net
+        1.1 If node is belpin then
+            1.1.1 Explore all belpins that are outputs for belpins that are used in design and store them in temp
+            1.1.2 Remove form temp belpins that are visible in child nodes
+            1.1.3 Add new found belpins to child nodes
+        1.2 Explore child nodes
+    2. Search for sources and sinks that go to the same BEL, if such pair exist, assume pseudo sitePIP
+        and connect them up into single tree
+Step 2 is only checked if net has more then 1 source
```

Comment:
Might be a typo, but a net can have multiple sinks, but only one source.

Reply:
Global nets can have multiple sources, and nets that were divided due to pseudo sitePIPs can have more then one source

Reply:
I think it is worth enhancing the comment to say which are the edge cases where a net can have multiple sources